### ISSUE TITLE
No regularization in the Keras xception implementation

### ISSUE THREAD

**Issue by guylinggl** [keras-team/keras#8721]
Hi all,
I went through the implementation of the xception network in Keras and it seems that the convolution layers regularization is set to the default which is None.  
https://github.com/fchollet/keras/blob/master/keras/applications/xception.py

This is inconsistent with the paper which states  a L2 regularization for the convolution layers.

https://arxiv.org/pdf/1610.02357.pdf

Am I missing something? 
Thanks!

**Comment by gabrieldemarmiesse** [comment]
Hi!

You are right! But when keras give the models, they're more intended to be used for fine tuning rather than redoing the experiments. As an example, vgg uses dropout, but the dropout layer isn't present in the keras implementation.
 My guess is that it can make the code base really complicated and not many people use keras models from applications to redo the experiments in the papers.

If you really want to, you can add the regularisation manually after loading the model. Just loop through the layers and do 
`for layer in model.layers:
    layer.W_regularizer = l2(...) `

Don't forget to compile afterwards, or your changes won't be used!

Hope this helps.

**Comment by guylinggl** [comment]
I guess that makes sense...
Thanks for your reply

**Comment by gabrieldemarmiesse** [comment]
No problem! Feel free to close the issue if all your questions hhave been answered.

Best regards.

**Comment by apatsekin** [comment]
I think it might be helpful to notice, that method proposed by @gabrieldemarmiesse won't affect loss of the model. Even after compilation. There might be other workarounds, but for now I just modify the source code of Xception to add `l2_reg`.

Some details: https://stackoverflow.com/questions/48330137/adding-regularizer-to-an-existing-layer-of-a-trained-model-without-resetting-wei/48330834#comment92783689_48330834
